servo/connectors/newrelic.py

```diff
@@ -0,0 +1,340 @@
+from __future__ import annotations
+import asyncio
+from collections import defaultdict
+from dateutil import parser as date_parser
+from functools import reduce
+from datetime import datetime, timedelta
+from typing import Awaitable, Callable, Dict, Iterable, List, Optional, Set, Tuple
+
+import httpx
+import httpcore._exceptions
+from pydantic import BaseModel, AnyHttpUrl, validator
+
+from servo import (
+    BaseChecks,
+    BaseConfiguration,
+    BaseConnector,
+    Control,
+    Check,
+    CheckHandler,
+    Description,
+    Duration,
+    Filter,
+    HaltOnFailed,
+    License,
+    Maturity,
+    Measurement,
+    Metric,
+    Unit,
+    check,
+    metadata,
+    on_event,
+    TimeSeries,
+    DurationProgress,
+)
+from servo.checks import create_checks_from_iterable, multicheck
+from servo.utilities import join_to_series
+
+
+DEFAULT_BASE_URL = 'https://api.newrelic.com'
+API_PATH = "/v2"
+
+# TODO: how are secrets handled?
+NEWRELIC_ACCOUNT_ID = str(open('/run/secrets/optune_newrelic_account_id').read()).strip()
+NEWRELIC_APM_API_KEY = str(open('/run/secrets/optune_newrelic_apm_api_key').read()).strip()
+NEWRELIC_APM_APP_ID = str(open('/run/secrets/optune_newrelic_apm_app_id').read()).strip()
+
+class NewrelicMetric(Metric):
+    """NewrelicMetric objects describe metrics that can be measure from the Newrelic apm api."""    
+    
+    fetch_name: str
+    """The name of the APM metric containing the values for this Metric
+
+    For details, see the [TODO better Newrelic resource](https://docs.newrelic.com/docs/apis/rest-api-v2/application-examples-v2/average-response-time-examples-v2) documentation.
+    """
+
+    values_selector: str
+    """
+    Values selector  of resultant time series
+    """
+
+    def __init__(self, name: str, unit: Unit, fetch_name: str, values_selector: str) -> None:
+        super().__init__(name=name, unit=unit, fetch_name=fetch_name, values_selector=values_selector, **kwargs)
+
+    def __check__(self) -> Check:
+        return Check(
+            name=f"Check {self.name}",
+            description=f"Run Newrelic get \"{self.fetch_name}: {self.values_selector}\""
+        ) # TODO Checker class
+
+# TODO: what is this used for?
+# class NewrelicTarget():
+#     pass # TODO
```

Comment:
I am guessing this is lifted from Prometheus. Targets in Prometheus refer to the instances being scraped. There's no equivalent in New Relic and this can be dropped.